applications/models.py

```diff
@@ -770,6 +778,71 @@ def get_all_occurrences(self):
             occurences[event_shedule.id] = event_shedule.get_occurences()
         return occurences
 
+    def create_aggregate_data(self):
+        total_events = []
+        total_events_durations = []
+        for schedule in self.application_event_schedules.all():
+            events_count = len(schedule.get_occurences().occurrences)
+            total_events.append(events_count)
+
+            time_dt_begin = datetime.timedelta(
+                hours=schedule.begin.hour, minutes=schedule.begin.minute
+            )
+            time_dt_end = datetime.timedelta(
+                hours=schedule.end.hour, minutes=schedule.end.minute
+            )
+            event_duration = time_dt_end - time_dt_begin
```

Comment:
I don't think begin and end can be used to determine duration, they are basically the start and end time of the "suitable" time range. So i think you should just get the duration from the application event min duration, which should be maybe good enough estimate.

Reply:
good point :thinking:  I'll change that

Reply:
this is changed